modules/nextflow/src/test/groovy/nextflow/config/parser/legacy/ConfigParserLegacyTest.groovy

```diff
@@ -739,6 +683,26 @@ class ConfigParserTest extends Specification {
 
     }
 
+    def 'should apply profiles in the order they were defined' () {
```

Comment:
@pditommaso this test is related to #1792 . We determined that config profiles are applied this way with the legacy parser. We could patch it, but I figure there's no point now, so just added a test to verify it's behavior.